### PR TITLE
Various Tweaks

### PR DESCRIPTION
**Pull request by jkreft-usgs**
These are some various small improvements to test coverage and best practices, plus some display bugs and changes

### CODE REVIEW

File: pubs_ui/pubswh/templates/pubswh/publication.html [hunk 1]
```diff
@@ -190,14 +190,31 @@ <h3 itemprop="name">{{ pubdata['title']|safe }}</h3>
                         {{- linkdisplaylists('Document','Document','Documents') -}}
                         {{- linkdisplaylists('Plate', 'Plate','Plates')  -}}
                         {{- linkdisplaylists('Sheet', 'Sheet','Sheets')  -}}
+                        {{- linkdisplaylists('Chapter','Chapter','Chapters') -}}
+                        {% if pubdata['interactions'] %}
+                            {% if pubdata['hasSubParts'] == True %}
```
Comment (ayan-usgs) [inline pubs_ui/pubswh/templates/pubswh/publication.html]: Would {% if pubdata['hasSubParts'] %} be sufficent here?

File: pubs_ui/pubswh/utils.py [hunk 2]
```diff
@@ -338,7 +338,13 @@ def get_pubs_search_results(self, params=None):
         :return: query results (or None) and response status code.
         :rtype: tuple
         """
-        search_result_obj = requests.get(url=self.search_url, params=params, verify=verify_cert)
+        # Only send the params if they contain information
+        if params is not None:
+            non_null_params = {}
+            for key, val in params.iteritems():
```
Comment (ayan-usgs) [inline pubs_ui/pubswh/utils.py]: You might be able to do this as a dictionary comprehension.

Comment (ayan-usgs) [inline pubs_ui/pubswh/utils.py]: `y = {k:v for (k, v) in x.iteritems() if v}`

Comment (jkreft-usgs) [inline pubs_ui/pubswh/utils.py]: fancy.

**Pull request merged**